### PR TITLE
refactor: label relocatable errors

### LINKED ISSUE #161: refactor: better labels for errors in `Relocatable` and `MaybeRelocatable`
At the moment, in both the `Relocatable` and `MaybeRelocatable` structures, we have errors that are not labeled properly using our error types that can be found in the `error.zig` file.
- Example in `Relocatable` structure: https://github.com/keep-starknet-strange/cairo-zig/blob/d50e5492e15d454f7882703df4824866543aa5a2/src/vm/memory/relocatable.zig#L114
- Example in `MaybeRelocatable` union: https://github.com/keep-starknet-strange/cairo-zig/blob/d50e5492e15d454f7882703df4824866543aa5a2/src/vm/memory/relocatable.zig#L391

We need to label these errors properly in order to be able to group them and enforce a better typing.

### PR DESCRIPTION
**Pull request by dhruvkelawala**
Closes #161

### CODE REVIEW

File: src/vm/memory/relocatable.zig [hunk 1]
```diff
@@ -420,7 +420,7 @@ pub const MaybeRelocatable = union(enum) {
     pub fn tryIntoRelocatable(self: Self) !Relocatable {
```
Comment (tcoratger) [inline src/vm/memory/relocatable.zig:420]: Here, as a result of your refactor, can we enforce a better typing of the output? like:

```zig
pub fn tryIntoRelocatable(self: Self) CairoVMError!Relocatable {
```

File: src/vm/memory/relocatable.zig [hunk 2]
```diff
@@ -126,7 +126,7 @@ pub const Relocatable = struct {
         other: u64,
     ) !Self {
```
Comment (tcoratger) [inline src/vm/memory/relocatable.zig:127]: Here, as a result of your refactor, can we enforce a better typing of the output? like:

```zig
pub fn subUint(
        self: Self,
        other: u64,
    ) MathError!Self {
```

File: src/vm/memory/relocatable.zig [hunk 3]
```diff
@@ -96,7 +96,7 @@ pub const Relocatable = struct {
     /// This method fails if `self` and other` are not from the same segment.
     pub fn sub(self: Self, other: Self) !Self {
```
Comment (tcoratger) [inline src/vm/memory/relocatable.zig:97]: Here, as a result of your refactor, can we enforce a better typing of the output?

File: src/vm/memory/relocatable.zig [hunk 4]
```diff
@@ -777,7 +777,7 @@ test "Relocatable: addFelt should add a relocatable and a Felt252" {
 
 test "Relocatable: addFelt should return an error if number after offset addition is too large" {
     try expectError(
-        error.ValueTooLarge,
+        CairoVMError.ValueTooLarge,
```
Comment (tcoratger) [inline src/vm/memory/relocatable.zig:780]: Can we move this one to a `MathError`?

File: src/vm/memory/relocatable.zig (hunk 3 above)
Comment (lana-shanghai) [reply src/vm/memory/relocatable.zig:97]: @tcoratger good point I will add a todo for that

Comment (lana-shanghai) [reply src/vm/memory/relocatable.zig:97]: ah there's more comments, OK let's wait for Dhruv to fix it

**Comment by dhruvkelawala** [comment]
So I have added stricter type checking for errors in most of the functions. In the remaining ones, I think we need a better error handling approach and cannot be fixed just by type checker. For example, 
https://github.com/dhruvkelawala/cairo-zig/blob/408cf6fb246af92d3d1c13eac344510ee2db6278/src/vm/memory/relocatable.zig#L98

This function can return CairoVMError or MathError but idts that Zig supports return either one of those. 

Something like `error{CairoVMError, MathError}!Self`. We can handle each error in this function separately and return proper errors. This should be done in a separate PR imo

**Review by tcoratger — APPROVED** [review]

**Pull request merged**